store/postgres/src/entities.rs

```diff
@@ -0,0 +1,993 @@
+use diesel::connection::SimpleConnection;
+use diesel::deserialize::QueryableByName;
+use diesel::dsl::{any, sql};
+use diesel::pg::{Pg, PgConnection};
+use diesel::sql_types::{Jsonb, Nullable, Text};
+use diesel::BoolExpressionMethods;
+use diesel::ExpressionMethods;
+use diesel::{debug_query, insert_into};
+use diesel::{OptionalExtension, QueryDsl, RunQueryDsl};
+use diesel_dynamic_schema::{schema, Column, Table as DynamicTable};
+use graph::data::subgraph::schema::SUBGRAPHS_ID;
+use graph::prelude::{
+    format_err, AttributeIndexDefinition, EntityChange, EntityChangeOperation, EntityFilter,
+    EntityKey, Error, EventSource, QueryExecutionError, StoreError, StoreEvent,
+    SubgraphDeploymentId, TransactionAbortError, ValueType,
+};
+use graph::serde_json;
+use inflector::cases::snakecase::to_snake_case;
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+use crate::filter::{build_filter, store_filter};
+use crate::functions::set_config;
+use crate::jsonb::PgJsonbExpressionMethods as _;
+
+/// Marker trait for tables that store entities
+pub(crate) trait EntitySource {}
+
+// The entities and related tables in the public schema. We put them in
+// this module to make sure that nobody else gets access to them. All access
+// to these tables must go through functions in this module.
+mod public {
+    table! {
+        entities (id, subgraph, entity) {
```

Comment:
For readability, lets put the primary key in the order it actually is.

Reply:
Done

Reply:
Actually, this was copied verbatim from `db_schema`, and when I change the order of pk columns to `entities (subgraph, entity, id)` I get a test failure:
```
thread 'reference_interface' panicked at 'assertion failed: `(left == right)`
  left: `"Object({\"leggeds\": List([Object({\"leg\": Null})])})"`,
 right: `"Object({\"leggeds\": List([Object({\"leg\": Object({\"id\": String(\"1\")})})])})"`', core/tests/interfaces.rs:180:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.


failures:
    reference_interface
``` 

I have no idea why that is; we should probably investigate that, but as part of a separate ticket. I'll leave that as is for now.

Reply:
Ah it must expect the columns in that order for `find` queries. Sure lets leave it as a refactor for later.